packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -2812,15 +2814,28 @@ function warnAboutUpdateOnUnmountedFiberInDEV(fiber) {
     } else {
       didWarnStateUpdateForUnmountedComponent = new Set([componentName]);
     }
-    console.error(
-      "Can't perform a React state update on an unmounted component. This " +
-        'is a no-op, but it indicates a memory leak in your application. To ' +
-        'fix, cancel all subscriptions and asynchronous tasks in %s.%s',
-      tag === ClassComponent
-        ? 'the componentWillUnmount method'
-        : 'a useEffect cleanup function',
-      getStackByFiberInDevAndProd(fiber),
-    );
+
+    // If we are currently flushing passive effects, change the warning text.
+    if ((executionContext & PassiveEffectContext) !== NoContext) {
+      console.error(
+        "Can't perform a React state update from within a passive effect destroy callback. " +
+          'To fix, move state updates to the useEffect() body in %s.%s',
+        tag === ClassComponent
```

Comment:
Is this even possible?

Reply:
Yes.

Reply:
I meant is it possible to get this warning with a class component? Referring to this particular line. The message seems confusing in that case.

Reply:
Oh I see. Nope. Only function components have passive effects.